From: Dev Jain <dev.jain@arm.com>
Date: Mon, 31 Aug 2026 08:28:47 +0000
Subject: [PATCH] mm/rmap: remove stale hugetlb check in try_to_unmap_one

Post commit d4ec5572825a ("mm/rmap: add
try_to_unmap_poisoned_hugetlb_one") try_to_unmap_one() cannot be called
with a hugetlb folio.  Therefore remove the folio_test_hugetlb() check.

Link: https://lore.kernel.org/20260831082849.3573957-1-dev.jain@arm.com
Signed-off-by: Dev Jain <dev.jain@arm.com>
Reviewed-by: Lorenzo Stoakes (ARM) <ljs@kernel.org>
Reviewed-by: Lance Yang <lance.yang@linux.dev>
Reviewed-by: Kunwu Chan <kunwu.chan@gmail.com>
Cc: David Hildenbrand <david@kernel.org>
Cc: Harry Yoo <harry@kernel.org>
Cc: Jann Horn <jannh@google.com>
Cc: Liam R. Howlett <liam@infradead.org>
Cc: Rik van Riel <riel@surriel.com>
Cc: Vlastimil Babka <vbabka@kernel.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---
 mm/rmap.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

diff --git a/mm/rmap.c b/mm/rmap.c
index fed0362e0bd0e..b5cc9273fe5a4 100644
--- a/mm/rmap.c
+++ b/mm/rmap.c
@@ -2297,11 +2297,8 @@ static bool try_to_unmap_one(struct folio *folio, struct vm_area_struct *vma,
 		VM_BUG_ON_FOLIO(!pvmw.pte, folio);
 
 		address = pvmw.address;
-		if (folio_test_hugetlb(folio)) {
-			pteval = huge_ptep_get(mm, address, pvmw.pte);
-		} else {
-			pteval = ptep_get(pvmw.pte);
-		}
+		pteval = ptep_get(pvmw.pte);
+
 		if (likely(pte_present(pteval))) {
 			pfn = pte_pfn(pteval);
 		} else {
-- 
2.47.3

